src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,51 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter,
+                                                      const OrderByMode ordered_by) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    using ColumnDataType = typename decltype(type)::type;
+
+    if constexpr (std::is_same_v<decltype(typed_segment), const ReferenceSegment&>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      auto segment_iterable = create_iterable_from_segment(typed_segment);
+      segment_iterable.with_iterators(position_filter, [&](auto segment_begin, auto segment_end) {
+        auto sorted_segment_search = detail::SortedSegmentSearch(
+            segment_begin, segment_end, ordered_by, _predicate_condition, type_cast_variant<ColumnDataType>(_value));
+
+        sorted_segment_search.scan_sorted_segment([&](auto begin, auto end) {
+          size_t output_idx = matches.size();
+
+          matches.resize(matches.size() + std::distance(begin, end));
```

Comment:
When I tested this for the regular search, it was actually harmful. Instead of growing exponentially, you now grow linearly.

Reply:
Your're totally right. The way we do it, it is just growing linearly.
So should we just drop the resize and use emplace_back for inserting the RowIDs or are there any better approaches?

Reply:
If your benchmarks show that the resize is beneficial, leave it in.